packages/web/src/References.tsx

```diff
@@ -0,0 +1,96 @@
+import * as Markprompt from '@markprompt/react';
+import React, { useCallback, useContext, useMemo } from 'react';
+
+const capitalize = (text: string) => {
+  return text.charAt(0).toUpperCase() + text.slice(1);
+};
+
+const removeFileExtension = (fileName: string) => {
+  const lastDotIndex = fileName.lastIndexOf('.');
+  if (lastDotIndex === -1) {
+    return fileName;
+  }
+  return fileName.substring(0, lastDotIndex);
+};
+
+type ReferenceProps = {
+  transformReferenceId?: (referenceId: string) => {
+    href: string;
+    text: string;
+  };
+  referenceId: string;
+  index: number;
+};
+
+const defaultTransformReferenceId = (referenceId: string) => ({
+  href: removeFileExtension(referenceId),
+  text: capitalize(removeFileExtension(referenceId.split('/').slice(-1)[0])),
+});
+
+const Reference = (props: ReferenceProps) => {
```

Comment:
Should be part of `@markprompt/react` (we want the same behavior for the references transformations, animations, etc).

Reply:
Here as well I think that any animations (and related code) should not be included in the headless components package. The presentation of loading state and references as they come in could be split up in different ways and represented in different places in the UI. For instance, I can image that implementers show a loading bar right under the prompt, or show the references as a sidebar. Therefore, I think this composition is not a headless component.

I do think we need to rewrite this component to be a bit more semantic and accessible, as just hiding paragraphs with opacity will not hide them from screen reader users. We can probably split up into separate loading and references components and use [ARIA Live Regions](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (set to `polite`) in order to improve this component.